private/bufpkg/bufconnect/static_token_provider.go

```diff
@@ -45,55 +36,48 @@ func NewTokenProviderFromString(token string) (TokenProvider, error) {
 // The special characters `:`, `@` and `,` are used as the splitters. The usernames, tokens, and
 // remote addresses does not contain these characters since they are enforced by the rules in BSR.
 func newTokenProviderFromString(token string, isFromEnvVar bool) (TokenProvider, error) {
+	// Tokens for different remotes are separated by `,`. Using strings.Split to separate the string into remote tokenToAuthKey.
+	tokens := strings.Split(token, ",")
+	if len(tokens) <= 1 {
+		return staticTokenProvider{
+			setBufTokenEnvVar: isFromEnvVar,
+			token:             token,
+		}, nil
+	}
 	tokenProvider := &staticTokenProvider{
 		setBufTokenEnvVar: isFromEnvVar,
-		tokenToAuthKey:    make(map[string]authKey),
+		keyPairs:          make(map[string]string),
 	}
-	// Tokens for different remotes are separated by `,`. Using strings.Split to separate the string into remote tokenToAuthKey.
-	tokens := strings.Split(token, ",")
 	for _, token := range tokens {
-		if keyPairs, remoteAddress, ok := strings.Cut(token, "@"); ok {
-			ak := authKey{}
-			if err := ak.unmarshalString(keyPairs); err != nil {
-				return nil, err
-			}
-			if _, ok = tokenProvider.tokenToAuthKey[remoteAddress]; ok {
-				return nil, fmt.Errorf("cannot parse token: %s, repeated token for same BSR remote: %s", token, remoteAddress)
-			}
-			tokenProvider.tokenToAuthKey[remoteAddress] = ak
-		} else {
-			if tokenProvider.defaultToken != "" {
-				return nil, fmt.Errorf("cannot parse token: two buf token provided: %q and %q", token, tokenProvider.defaultToken)
-			}
-			tokenProvider.defaultToken = token
+		key, hostname, found := strings.Cut(token, "@")
```

Comment:
`Cut` will only split into two, no matter how many `@`s there are. You want to verify there is exactly one.

```
split := strings.Split(token, "@")
if len(split) != 2 {
  return nil, fmt.Errorf("...")
}
```

Reply:
you should make sure that both sides of the split are non-empty as well